core/store/migrations/migration1554855314/migrate.go

```diff
@@ -0,0 +1,84 @@
+package migration1554855314
+
+import (
+	"github.com/jinzhu/gorm"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/store/migrations/migration1551816486"
+	"github.com/smartcontractkit/chainlink/core/store/orm"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+// Migration is the singleton type for this migration
+type Migration struct{}
+
+// Migrate adds the sync_events table
+func (m Migration) Migrate(tx *gorm.DB) error {
+	tx = tx.Begin()
+	if err := tx.Exec(`
+CREATE TABLE "bridge_types_with_incoming_token_hash" (
+	"name" varchar(255),
+	"url" varchar(255),
+	"confirmations" bigint,
+	"incoming_token_hash" VARCHAR(64),
+	"salt" VARCHAR(32),
+	"outgoing_token" varchar(255),
+	"minimum_contract_payment" varchar(255),
+	PRIMARY KEY ("name")
+);`).Error; err != nil {
+		tx.Rollback()
+		return errors.Wrap(err, "error creating temporary bridge_types_with_incoming_token_hash table")
+	}
+
+	// CAUTION: This could be quite slow with a big enough database
+	if err := orm.Batch(1000, func(offset, limit uint) (uint, error) {
+		var bts []migration1551816486.BridgeType
+		err := tx.Limit(limit).Offset(offset).Find(&bts).Error
+		if err != nil {
+			return 0, errors.Wrap(err, "error loading old bridge types")
+		}
+
+		for _, bt := range bts {
+			newBT := struct {
```

Comment:
Maybe break this out into a named type so that we can refer to it in the subsequent migration that touches BridgeType, the way you did with the last version of BridgeType? That way we won't have to touch past migrations in the future.